poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -91,16 +97,19 @@ impl Extend<Heartbeat> for Heartbeats {
             if heartbeat.validity != proto::HeartbeatValidity::Valid {
                 continue;
             }
-            self.heartbeats.insert(
-                HeartbeatKey {
+            let entry = self
+                .heartbeats
+                .entry(HeartbeatKey {
                     hotspot_key: heartbeat.hotspot_key,
                     cbsd_id: heartbeat.cbsd_id,
-                },
-                HeartbeatValue {
-                    reward_weight: heartbeat.reward_weight,
-                    timestamp: heartbeat.timestamp,
-                },
-            );
+                })
+                .or_default();
+            entry.reward_weight = heartbeat.reward_weight;
+            if entry.timestamps.is_empty()
+                || entry.timestamps[0].hour() != heartbeat.timestamp.hour()
```

Comment:
maybe overkill but why not take the difference in timestamps and insert if it's equal to a duration >= 1 hour? i'm picturing a radio that's been offline for 12 hours and registering its next heartbeat in the same "hour" as the last recorded one but i suppose if they're consistently online they should never go so long between successful heartbeats that would be a likely scenario

Reply:
It was easier to do it this way, and again this is only used for the generate cli